docs/bsp.md

```diff
@@ -1207,6 +1208,90 @@ Clients can use this information to reconstruct the tree of sbt meta builds. The
 `parent` information can be defined from `children` but it's provided by the server to
 simplify the data processing on the client side.
 
+## Connecting to a BSP Server
+
+The Build Server Protocol defines a standard to connect to BSP servers.
+This standard has been designed such that:
+
+1. Clients don't need to have specific support for a server to be able to
+   connect to it.
+1. Clients can connect to build tools installed both at the machine or at the
+   project level.
+1. Different versions of the same build tool can have different connection
+   mechanisms.
+
+The following protocol to discover and connect to BSP servers is inspired by
+[Jupyter's
+Kernelspec](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs)
+and incorporates changes required to deal with  requirements above.
+
+### The BSP Connection Details
+
+The BSP connection details are modelled in a JSON object generated by a build
+tool command. The object contains the following fields to allow clients to
+connect to the server.
+
+```scala
+trait BspConnectionDetails {
```

Comment:
We need to specify some argument variable patterns that a client can pass to the server invocation for things like 
* socket file
* port
* bsp protocol version(?)
* flags (?)

Reply:
Yes, this is a critical part, good point. I'm still rumbling how to support this in a principled way. I'll comment on it tonight.

Reply:
That’s exactly what I wanted to avoid, the connection file was in the original bsp ticket, but it looks like we’ll need it after all 😄